activiti-cloud-services-org/activiti-cloud-services-org-service/src/main/java/org/activiti/cloud/services/organization/validation/extensions/ProcessExtensionsModelValidator.java

```diff
@@ -0,0 +1,139 @@
+import org.activiti.cloud.organization.api.ValidationContext;
+import org.activiti.cloud.organization.api.process.Extensions;
+import org.activiti.cloud.organization.converter.JsonConverter;
+import org.activiti.cloud.organization.core.error.ModelingException;
+import org.activiti.cloud.organization.core.error.SemanticModelValidationException;
+import org.activiti.cloud.organization.core.error.SyntacticModelValidationException;
+import org.activiti.cloud.services.organization.converter.BpmnProcessModelContent;
+import org.activiti.cloud.services.organization.converter.ProcessModelContentConverter;
+import org.activiti.cloud.services.organization.validation.JsonSchemaModelValidator;
+import org.everit.json.schema.loader.SchemaLoader;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.stereotype.Component;
+
+import static java.lang.String.format;
+import static org.activiti.cloud.services.common.util.ContentTypeUtils.CONTENT_TYPE_JSON;
+import static org.apache.commons.lang3.StringUtils.removeStart;
+
+@Component
+@ConditionalOnMissingBean(name = "ProcessExtensionsModelValidator")
+public class ProcessExtensionsModelValidator extends ExtensionsJsonSchemaValidator {
+
+    public static final String UNKNOWN_PROCESS_ID_VALIDATION_ERROR_PROBLEM = "Unknown process id in process extensions: %s";
+    public static final String UNKNOWN_PROCESS_ID_VALIDATION_ERROR_DESCRIPTION = "The process extensions are bound to an unknown process id '%s'";
+
+    private final SchemaLoader processExtensionsSchemaLoader;
+
+    private final Set<ProcessExtensionsValidator> processExtensionsValidators;
+
+    private final ProcessModelType processModelType;
+
+    private JsonConverter<Extensions> jsonExtensionsConverter;
+
+    private final ProcessModelContentConverter processModelContentConverter;
+
+    @Autowired
+    public ProcessExtensionsModelValidator(SchemaLoader processExtensionsSchemaLoader,
+                                    Set<ProcessExtensionsValidator> processExtensionsValidators,
+                                    ProcessModelType processModelType,
+                                    JsonConverter<Extensions> jsonExtensionsConverter,
+                                    ProcessModelContentConverter processModelContentConverter) {
+        this.processExtensionsSchemaLoader = processExtensionsSchemaLoader;
+        this.processExtensionsValidators = processExtensionsValidators;
+        this.processModelType = processModelType;
+        this.jsonExtensionsConverter = jsonExtensionsConverter;
+        this.processModelContentConverter = processModelContentConverter;
+    }
+
+    @Override
+    protected Stream<ModelValidationError> validateModelExtensions(Model model,
+                                                                   ValidationContext context) {
+        return Optional.ofNullable(model.getId())
```

Comment:
would not without the stream make the code easier to understand in this specific case? Why returning a stream if we expect a limited number of results?

Reply:
Ok, changed to list and renamed so it is easier to understand the code